lib/user_name_suggester.rb

```diff
@@ -110,6 +110,14 @@ def self.sanitize_username(name)
 
     if SiteSetting.unicode_usernames
       name.unicode_normalize!
+
+      # TODO: Jan 2022, review if still needed
+      # see: https://meta.discourse.org/t/unicode-username-with-as-the-final-char-leads-to-an-error-loading-profile-page/173182
+      if name.include?('Σ')
+        ctx = MiniRacer::Context.new
+        name = ctx.eval("#{name.inspect}.toLowerCase()")
```

Comment:
This is probably OK, but I did worry for a second that you are passing `name` in to `eval`. It looks like `inspect` should catch any escaping of the string right?

Could we do this after the `gsub`s below? Those would remove any quotes and things.

Reply:
I don't trust ruby's escaping to match JS's. This should be passed through a jsescape library function instead of `#inspect`.

Reply:
ok I changed it to `to_json` this certainly makes is far clearer. json is cross compatible so we are unconditionally secure here.

I am not sure what the specific issues are with inspect, but agree that it is not impossible there are some cross compatibility concerns